File: generated-code/php/codecraft/FileReadWrite/Model/DebugInterface/PrimitiveType.php
<?php

namespace Model\DebugInterface {
    /**
     * Primitive type for debug rendering
     */
    abstract class PrimitiveType
    {
        /**
         * Lines, number of vertices should be divisible by 2
         */
        const LINES = 0;

        /**
         * Triangles, number of vertices should be divisible by 3
         */
        const TRIANGLES = 1;

        /**
         * Read PrimitiveType from input stream
         */
        public static function readFrom($stream)
        {
            $result = $stream->readInt32();
            if (0 <= $result && $result < 2) {
                return $result;
            }
            throw new Exception('Unexpected tag value');
        }
    }
}

File: generated-code/csharp/codecraft/FileReadWrite/Model/DebugInterface/PrimitiveType.cs
namespace TransGenTest.Model.DebugInterface
{
    /// <summary>
    /// Primitive type for debug rendering
    /// </summary>
    public enum PrimitiveType
    {
        /// <summary>
        /// Lines, number of vertices should be divisible by 2
        /// </summary>
        Lines = 0,
        /// <summary>
        /// Triangles, number of vertices should be divisible by 3
        /// </summary>
        Triangles = 1,
    }

    public static class PrimitiveTypeHelper {
        /// <summary> Read PrimitiveType from reader </summary>
        public static PrimitiveType ReadFrom(System.IO.BinaryReader reader) {
            switch (reader.ReadInt32())
            {
                case 0:
                    return PrimitiveType.Lines;
                case 1:
                    return PrimitiveType.Triangles;
                default:
                    throw new System.Exception("Unexpected tag value");
            }
        }
    }
}